homeassistant/components/sensor/filter.py

```diff
@@ -137,7 +146,21 @@ def filter_sensor_state_listener(entity, old_state, new_state):
                 self._unit_of_measurement = new_state.attributes.get(
                     ATTR_UNIT_OF_MEASUREMENT)
 
-            self.async_schedule_update_ha_state()
+            if update_ha:
+                self.async_schedule_update_ha_state()
+
+        if self._history_period and 'recorder' in self.hass.config.components:
+            start = dt_util.utcnow() - self._history_period
+
+            history_list = await self.hass.async_add_job(partial(
+                history.state_changes_during_period, self.hass,
+                start, entity_id=self._entity))
+
+            prev_state = None
+            for state in history_list[self._entity]:
+                filter_sensor_state_listener(
```

Comment:
I'm not sure, but this might not work too well with #13104; because in there you're using `now = int(dt_util.utcnow().timestamp())` even though the state changes are from the past. I think you could use the existing [`State#last_updated`](https://github.com/home-assistant/home-assistant/blob/5a9013cda58a3a99e1f1e9fa73e18c5895a4f56c/homeassistant/core.py#L518) field.

Reply:
Yes, the two PR's started at different times and will now require adequate merging